### ISSUE TITLE
Design System v3: `ModelViewer.astro` für optionale 3D-Elemente

### ISSUE THREAD

**Issue by LuisArteaga** [LuisArteaga/speakdatawith.com#23]
## Voraussetzungen

- Issues „Design System v1" und „v2" sind gemergt: Tokens, `BrandOwl`, `Badge`, `Button`, `ContentCard`, `Hero` (mit `visual`-Slot) und `SectionHeading` existieren.
- Das 3D-Asset liegt im Repo unter `public/models/` – **verwende ausschließlich dieses Asset:**

> **⚠️ Vom Issue-Ersteller vor dem Absenden einzutragen:**
> - Modell-Datei: `public/models/<name>.glb`
> - Poster-Bild: `public/models/<name>-poster.webp` (statisches Vorschaubild; falls nicht vorhanden, soll der Agent es nicht erfinden, sondern einen einfarbigen Token-basierten Platzhalter mit Owl-SVG rendern)
> - Alt-Text: `<beschreibender Text>`

## Arbeitsweise

Beginne nicht sofort mit der Implementierung. Analysiere zuerst die v1/v2-Komponenten und die Zielseite. Poste einen kurzen Plan als Kommentar: Lade-Strategie, Bundle-Auswirkung, betroffene Dateien. Implementiere danach.

## Ziel

Eine isolierte, wiederverwendbare `ModelViewer.astro`-Komponente, die auf ausgewählten Seiten interaktive 3D-Modelle darstellt – ohne die restliche, statische Website zu beeinträchtigen.

## Entscheidungen (nicht verhandelbar)

- Bibliothek: `@google/model-viewer` als npm-Dependency – **kein CDN-Import**
- Dies ist die **einzige** Komponente der Website, die Client-JavaScript laden darf
- Das JS wird nur auf Seiten geladen, die `ModelViewer` tatsächlich verwenden
- `Hero.astro` wird **nicht verändert** – die 3D-Nutzung erfolgt ausschließlich über den vorhandenen `visual`-Slot
- Kein Astronaut, keine externen Demo-Modelle, keine Modelle von fremden URLs

### Props

| Prop | Typ | Pflicht | Beschreibung |
|---|---|---|---|
| `src` | string | ja | Pfad zum `.glb` unter `/models/` |
| `alt` | string | ja | Beschreibender Alternativtext |
| `poster` | string | nein | Pfad zum Poster-Bild |
| `caption` | string | nein | Monospace-Label unter dem Viewer (Stil wie `ContentCard`-Caption) |
| `height` | number | nein | Höhe in px, Default `320` |
| `autoRotate` | boolean | nein | Default `true`; wird bei `prefers-reduced-motion: reduce` immer deaktiviert |

### Lade-Strategie

1. Der Viewer rendert serverseitig einen Container mit **fester Höhe** (`height`-Prop) und dem Poster bzw. Platzhalter – dadurch **kein Layout Shift**.
2. Die `model-viewer`-Bibliothek wird erst geladen, wenn der Container in den Viewport kommt (`IntersectionObserver`, danach dynamischer `import('@google/model-viewer')`).
3. Bis dahin und bei einem Ladefehler bleibt das Poster/der Platzhalter sichtbar – die Seite ist ohne 3D vollständig nutzbar.
4. `interaction-prompt="none"`, `camera-controls` aktiv, `shadow-intensity="1"`, `environment-image="neutral"` (Werte aus der Moodboard-Referenz).

### Styling

- Der Viewer wird typischerweise innerhalb einer `ContentCard` verwendet – keine eigene Card-Optik duplizieren
- Transparenter Viewer-Hintergrund (`--poster-color: transparent`)
- Alle Farben/Radien aus Tokens; keine Hex-Werte

### Barrierefreiheit & Motion

- `alt` ist Pflicht-Prop; fehlt sie, soll der Build fehlschlagen (Astro-Props-Typisierung) oder eine klare Fehlermeldung erscheinen
- `prefers-reduced-motion: reduce`: kein Auto-Rotate; Kamera-Interaktion per Nutzereingabe bleibt erlaubt
- Der Viewer darf keine Tastaturfalle erzeugen; Fokus-Verhalten der Web Component nicht überschreiben

## Einbindung

Binde den `ModelViewer` auf **genau einer Seite** ein:

> **⚠️ Vom Issue-Ersteller anzupassen:** z. B. `src/pages/index.astro`, im `visual`-Slot des Heros:

```astro
<Hero title="…" description="…" ctaText="…" ctaHref="…">
  <ContentCard slot="visual" caption="[CAD.3D_MODEL_INTERACTIVE]">
    <ModelViewer src="/models/<name>.glb" alt="<Alt-Text>" poster="/models/<name>-poster.webp" />
  </ContentCard>
</Hero>
```

Alle anderen Seiten bleiben unverändert und JS-frei.

## Performance-Budget

- `.glb`-Datei ≤ 3 MB (liegt bereits im Repo; falls größer: im Plan-Kommentar melden statt selbst komprimieren)
- Die `model-viewer`-Bibliothek erscheint **nicht** im initialen Bundle von Seiten ohne `ModelViewer`
- Poster/Platzhalter ≤ 100 KB

## Nicht-Ziele

- Keine Änderung an `Hero.astro`, `BrandOwl.astro` oder anderen v1/v2-Komponenten
- Kein Three.js, kein React-Three-Fiber, kein eigenes WebGL
- Keine 3D-Modelle generieren, konvertieren oder von externen Quellen laden
- Kein `model-viewer` im globalen Layout oder in `BaseLayout`
- Keine weiteren Seiten mit 3D ausstatten
- Keine Fallback-„Simulation" des 3D-Modells per CSS-Animation

## Akzeptanzkriterien (prüfbar)

- [ ] `npm run build` läuft fehlerfrei durch
- [ ] `@google/model-viewer` ist die einzige neue Dependency in `package.json`
- [ ] `grep -rn "modelviewer.dev\|ajax.googleapis" src/` liefert keine Treffer (kein CDN, kein Demo-Asset)
- [ ] `grep -rn "Astronaut" src/ public/` liefert keine Treffer
- [ ] `ModelViewer.astro` existiert mit den beschriebenen Props; `src` und `alt` sind als Pflicht typisiert
- [ ] Im Build-Output (`dist/`) laden nur Seiten mit `ModelViewer` das zugehörige Script – prüfbar via grep auf den Script-Chunk in den HTML-Dateien anderer Seiten
- [ ] Der Container hat eine feste Höhe vor dem JS-Load (kein CLS)
- [ ] `prefers-reduced-motion` deaktiviert Auto-Rotate
- [ ] Bei deaktiviertem JavaScript zeigt die Seite Poster/Platzhalter und bleibt vollständig nutzbar
- [ ] `Hero.astro` ist unverändert (git diff)

## Erwartete Lieferung

1. Implementierung als PR
2. Plan-Kommentar vor Implementierungsbeginn
3. Nachweis der Lade-Strategie: welcher Chunk wird wann geladen, Auswirkung auf Seiten ohne 3D
4. Build-Output als Nachweis
5. Ergänzung von `ModelViewer` in `docs/design-system.md` inkl. Beispiel im `visual`-Slot und Hinweis auf das Performance-Budget
